src/pymovements/dataset/_utils/_downloads.py

```diff
@@ -198,9 +231,120 @@ def _download_url(url: str, destination: Path, verbose: bool = True) -> None:
     verbose : bool
         If True, show progressbar.
     """
+    # Preflight request to avoid urlretrieve creating an HTTPError with
+    # unclosed temporary file objects on Python 3.14 (PytestUnraisableExceptionWarning).
+    _raise_if_http_error(url)
+
+    # Stream download with an opener that does NOT raise HTTPError automatically.
+    # This prevents creation of HTTPError objects that can hold temp files and
+    # lead to unraisable warnings on Python 3.14.
+    opener = _build_no_http_error_opener()
+    req = urllib.request.Request(
+        url,
+        headers={'User-Agent': USER_AGENT, 'Accept': 'application/octet-stream'},
+    )
     with _DownloadProgressBar(desc=destination.name, disable=not verbose) as t:
-        urllib.request.urlretrieve(url=url, filename=destination, reporthook=t.update_to)
-        t.total = t.n
+        resp = None
+        try:
```

Comment:
the body of the try is so big that it's not really clear which functions are actually raising the exceptions.

please try to keep the logic inside the try block minimal.